dev/com.ibm.ws.microprofile.health/src/com/ibm/ws/microprofile/health/internal/HealthCheckHttpResponseBuilder.java

```diff
@@ -112,28 +114,34 @@ protected void setChecks(HealthCheckResponse response) {
             overallState = State.DOWN; // treat as fail case
         }
 
-        Optional<Map<String, Object>> data = response.getData();
-        if ((data != null) && data.isPresent())
-            check.put(HEALTH_CHECK_PAYLOAD_DATA, data.get());
+        Optional<Map<String, Object>> dataMap = response.getData();
+        if ((dataMap != null) && dataMap.isPresent()) {
+            JsonObjectBuilder data = jsonBuilderFactory.createObjectBuilder();
+            for (Map.Entry<String, Object> entry : dataMap.get().entrySet()) {
+                System.out.println("HERE2!");
+
+                data.add(entry.getKey(), entry.getValue().toString());
+                System.out.println("HERE3!" + entry.getValue().toString());
```

Comment:
I assume these print outs will be removed and similar comment as above with value being null would cause a NullPointerException

Reply:
sigh, oops